tests/src/test/scala/whisk/core/database/test/DatabaseScriptTestUtils.scala

```diff
@@ -0,0 +1,114 @@
+    val dbUrl = s"${dbProtocol}://${dbUsername}:${dbPassword}@${dbHost}:${dbPort}"
+
+    /** Creates a new database with the given name */
+    def createDatabase(name: String, designDocPath: Option[String])(implicit as: ActorSystem, logging: Logging) = {
+        // Implicitly remove database for sanitization purposes
+        removeDatabase(name, true)
+
+        println(s"Creating database: $name")
+        val db = new ExtendedCouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, name)
+        retry({ db.createDb().futureValue shouldBe 'right }, N = 10, waitBeforeRetry = Some(500.milliseconds))
+
+        retry({
+            val list = db.dbs().futureValue.right.get
+            list should contain(name)
+        }, N = 10, waitBeforeRetry = Some(500.milliseconds))
+
+        designDocPath.map { path =>
+            val designDoc = Source.fromFile(path).mkString.parseJson.asJsObject
+            db.putDoc(designDoc.fields("_id").convertTo[String], designDoc).futureValue
+        }
+
+        db
+    }
+
+    /** Wait for database to appear */
+    def waitForDatabase(dbName: String)(implicit as: ActorSystem, logging: Logging) = {
+        val client = new ExtendedCouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, dbName)
+        waitfor(() => {
+            client.getAllDocs(includeDocs = Some(true)).futureValue.isRight
+        })
+        client
+    }
+
+    /** Removes the database with the given name */
+    def removeDatabase(name: String, ignoreFailure: Boolean = false)(implicit as: ActorSystem, logging: Logging) = {
+        println(s"Removing database: $name")
+        val db = new ExtendedCouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, name)
+        retry({
+            val delete = db.deleteDb().futureValue
+            if (!ignoreFailure) delete shouldBe 'right
+        }, N = 10, waitBeforeRetry = Some(500.milliseconds))
+        db
+    }
+
+    /** Wait for a document to appear */
+    def waitForDocument(client: ExtendedCouchDbRestClient, id: String) = waitfor(() => client.getDoc(id).futureValue.isRight)
+
+    /** Get all docs within one database */
+    def getAllDocs(dbName: String)(implicit as: ActorSystem, logging: Logging) = {
+        val client = new ExtendedCouchDbRestClient(dbProtocol, dbHost, dbPort.toInt, dbUsername, dbPassword, dbName)
+        val documents = client.getAllDocs(includeDocs = Some(true)).futureValue
+        documents shouldBe 'right
+        documents.right.get
+    }
+
+    /** wait until all documents are processed by the view */
+    def waitForView(db: CouchDbRestClient, designDoc: String, viewName: String, numDocuments: Int) = {
+        waitfor(() => {
+            val view = db.executeView(designDoc, viewName)().futureValue
+            view shouldBe 'right
+            view.right.get.fields("rows").convertTo[List[JsObject]].length == numDocuments
+        }, totalWait = 2.minutes)
```

Comment:
Markus suggested above to create one local alias for retry, to pass only the task parameter. As result all `waitBeforeRetry` parameters are the same in this test.
If we use `retry` for the "longrunning" cloudant tasks as well, we have to increase this parameter for all retries or use the original one.
I decided to keep the `waitfor` for longrunning tasks and `retry` for the requests that should be there immediately.